keel-core/src/main/kotlin/com/netflix/spinnaker/keel/jackson/KeelApiModule.kt

```diff
@@ -63,6 +70,14 @@ object KeelApiModule : SimpleModule("Keel API") {
       setMixInAnnotations<SubnetAwareRegionSpec, SubnetAwareRegionSpecMixin>()
       setMixInAnnotations<Resource<*>, ResourceMixin>()
       setMixInAnnotations<ResourceSpec, ResourceSpecMixin>()
+      insertAnnotationIntrospector(object : NopAnnotationIntrospector() {
+        override fun findCreatorAnnotation(config: MapperConfig<*>, a: Annotated): JsonCreator.Mode? =
+          when {
+            a.hasAnnotation(Factory::class.java) -> PROPERTIES
+            a.rawType.constructors.any { it.isAnnotationPresent(Factory::class.java) } -> DISABLED
```

Comment:
Do you mind explaining what these two cases account for?

Reply:
First is "does this constructor have a `@Factory` annotation? If so, use it for constructing the object".
Second is "does _any other_ constructor have a `@Factory` annotation? If so, _do not_ use _this_ one".

Without that second case `@Factory` will conflict with the default constructor and an exception gets thrown.

Reply:
I'll add a comment to that effect

Reply:
I actually meant to pull this out as a proper class instead of an anonymous object too.